gobblin-runtime/src/main/java/gobblin/scheduler/JobScheduler.java

```diff
@@ -404,46 +402,44 @@ private void scheduleGeneralConfiguredJobs()
    *   is called on the changes is not controlled by Gobblin, but instead by the monitor itself.
    * </p>
    */
-  private void startJobConfigFileMonitor()
+  private void startGeneralJobConfigFileMonitor()
       throws Exception {
-    LOG.info("Monitor debugging: " + this.properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_DIR_KEY));
-    final File jobConfigFileDir = new File(this.properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_DIR_KEY));
-    FileAlterationListener listener = new FileAlterationListenerAdaptor() {
-      /**
-       * Called when a new job configuration file is dropped in.
-       */
+    final Path jobConfigFileDirPath =
+        new Path(this.properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY));
+    PathAlterationListener listener = new PathAlterationListenerAdaptor() {
```

Comment:
Can you refactor this into an actual class (we shouldn't use long anonymous classes)